go/pserver/service.go

```diff
@@ -49,13 +53,58 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// Checkpoint saves the checkpoint for pserver
+type Checkpoint struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+	State     []byte
+	ParameterWithConfig
```

Comment:
Agree with your comment on the other PR, `Checkpoint` need to contain checkpoints for all parameters on the pserver.
I was confused before. I think `Checkpoint` should be renamed to `CheckpointInfo`, it's actually a reference to the checkpoint file on distributed filesystem. The type of `Timestamp` should be changed to `int64`.
Perhaps something like:
```go
type CheckpointInfo struct {
  UUID string
  Path string
  MD5 string
  Timestamp int64
}
```

Maybe we can have another type like:
```go
type ParameterCheckpoint struct {
  ParameterWithConfig
  State []byte
}

type Checkpoint []ParameterCheckpoint
```

How do you think?